app/views/portal/show.rb

```diff
@@ -783,154 +753,173 @@ def iiif_manifesto(identifier, collection)
-          # - full image needed for the others
-          # - metadata service needed
-          # if web_resource_url != edm_resource_url
-          #   item[:thumbnail] = web_resource_url
-          # end
-        elsif media_type == 'audio' || media_type == 'sound'
-          item['is_audio'] = true
-          players << { audio: true }
-          item[:thumbnail] = item[:thumbnail] || 'http://europeanastatic.eu/api/image?size=BRIEF_DOC&type=SOUND'
-        elsif media_type == 'iiif'
-          item['is_iiif'] = true
-          players << { iiif: true }
-        elsif media_type == 'pdf'
-          item['is_pdf'] = true
-          players << { pdf: true }
-        elsif media_type == 'text'
-          if mime_type == 'application/pdf'
-            item['is_pdf'] = true
-            players << { pdf: true }
-          else
-            item['is_text'] = true
-          end
-        elsif media_type == 'video'
-          item['is_video'] = true
-          players << { video: true }
-        else
-          item['is_unknown_type'] = render_document_show_field_value(web_resource, 'about')
-        end
+    def item_downloadable?(web_resource_url, mime_type, media_type, media_rights)
+      if web_resource_url.blank? ||
+        mime_type.blank? ||
+        download_disabled?(media_rights) ||
+        (media_type == 'text' && mime_type == 'text/plain; charset=utf-8') ||
+        (media_type == 'video' && mime_type == 'text/plain; charset=utf-8')
+        false
+      else
+        true
+      end
+    end
 
-        if mime_type == 'application/pdf' || mime_type == 'audio/flac'
-          item['play_url'] = edm_is_shown_by_download_url
-        elsif !manifesto.nil?
-          item['play_url'] = manifesto
-          item[:playable] = true
-        else
-          item['play_url'] = web_resource_url
-        end
+    def item_thumbnail(media_type)
+      edm_type = (media_type == 'audio' ? 'SOUND' : media_type.upcase)
+      'http://europeanastatic.eu/api/image?size=BRIEF_DOC&type=' + edm_type
+    end
 
-        # TODO: this should check the download-ability of the web resource
-        if edm_is_shown_by_download_url.present?
-          item['download'] = {
-            url: edm_is_shown_by_download_url,
-            text: t('site.object.actions.download')
-          }
-          item['technical_metadata'] = {
-            mime_type: mime_type
-          }
-        end
+    # Media type function normalises mime types
+    def media_type(mime_type)
+      case (mime_type || '').downcase
+      when /^audio\//
+        'audio'
+      when /^image\//
+        'image'
+      when /^video\//
+        'video'
+      when /^text\//, /\/pdf$/
+        'text'
+      end
+    end
 
-        # make sure the edm_is_shown_by is the first item
-        if web_resource_url == edm_resource_url
-          item[:is_current] = true
-          items.unshift(item)
-        else
-          items << item
+    def item_player(media_type, mime_type)
```

Comment:
Cyclomatic complexity for item_player is too high. [8/6]<br>Method has too many lines. [14/10]